### ISSUE TITLE
About compatibility

### ISSUE THREAD

**Issue by Th3MadHatter** [Kaldaien/UnX#55]
First, I love your mod, its truly awesome, but.

Are you seriously dropping Win 7 all together?
With every new release I see "won't work on win 7, I don't care".

There are so many users that simply can't install Win 8 or 10 due to hardware issues, I can't upgrade because over several pieces of equipment I use for work on ALL my devices, and I'm not even complaining about the software issues, incompatibility problems, bugs and major issues. I had to roll back 3 times already, once from win 8 and twice from win 10 just to get everything back in working order so I simply can't use newer clients unless they drastically change things, which isn't going to happen.
I'm not even going to point at the privacy issues both versions have, those are well known and, in my opinion, not that important but for a lot of users they are so also keep that in mind, turning off a few sliders in the settings does not make you bullet proof from snooping and accidental leaks to clouds and such, which can be a major issue for those that work with sensitive information on Windows PC's (they shouldn't in general but most people simply do because they don't have the knowledge to run other OSes that are *NIX based or such).

Long story short, could you at least mull it over and try and reconsider your "do I care if it works on win 7, probably not" statement?
A lot of users will stick with Win 7 for the upcoming years, I am one of them but I know a lot of people that are as well, just based on software and hardware needs that the architecture doesn't support. Win 7 is not outdated and won't be for a while.

Can you say, "sc3w them, update or just don't use my mod" sure you can but its github, you build things to help people that can't do so them self (right?).

Please consider some love for Win 7 users, I would love to use the new features in your mod but I'm stuck in the v5 series just because of this. Just some backward capability, even if we have to run a version or so behind on the main branch.

Thanks for your hard work on UnX so far, it really is awesome.

**Comment by Kaldaien** [comment]
Yes, I am getting rid of Windows 7 compatibility. It is an absolute nightmare to work with that operating system, and I simply do not have time for it anymore.

**Comment by Th3MadHatter** [comment]
Then I will just have to hope for it to work at some point or stay below v6.xx, maybe some else wants to fork it or make pulls making it win 7 compatible if you don't feel like supporting it anymore. 
I would do it myself but I have no experience with the language you use so I'd have to learn an entire language for it.

Please do keep in mind that, right now over 60% of home users still use Win 7 and that roughly 45% will continue using it. Roughly 5% will upgrade, the rest will either migrate to OSX or Linux so that is 45% of the Windows users that will keep using the Win 7 platform because of various hardware and software issues etc. that simply will not get fixed in an 8+ environment until maybe Win 11 is released.

These are just basic statistics, I'm not even calculating the people that are currently using Win 10 but are opting for a rollback to Win 7, not because they are hardcore win 7 users but simply because it doesn't support their needs, e.g. hardware/software not preforming as it should, the lack of user control Win 8+ offers users, e.g. the doubled up strain Win 10 provides to graphic rendering, and the severe lack of privacy (which is great for multi-platform use on phones and tablets but a lot less for actual desktop/laptops).

Doesn't whatever program you use to code in allow you to test in different OSes? 
I'm used to multi-platform coding in different languages but all of the software I've used so far is able to mimic nearly every platform.
If it really doesn't however, you can simply set up a virtual machine for it, that's a full proof way to do multi-platform testing in live environments, it doesn't even take up much disk space because you simply require the core and some basic GUI.

Also, doesn't 8+ have a compatibility mode? It's a so-so way to troubleshoot but it has been useful for me in the past.

**Comment by Kaldaien** [comment]
No, Visual Studio doesn't have a simulator for Windows 7. The only stuff I've ever worked with that have simulators are multi-thousand dollar embedded systems :)

What Microsoft does provide is an Application Verifier, which basically looks at all of the application's activity while it is running and verifies it for correctness against a set of rules. I have set the strictest possible rules for Windows NT 6.1 (Windows 7) and it passes. Meanwhile, on an actual Windows 7 system it deadlocks immediately.

So you can see why I don't have any faith in that and why I don't like Windows 7 to begin with. As a graphics programmer, half of D3D11 isn't even supported on Windows 7 and nothing stops you from accidentally straddling the line between what is and is not supported aside from crashing. It's Windows Vista 1.1; a terrible platform that needs to go away :)

I'll get the software working on Windows 7 again eventually, but I'm not dedicated to keeping it working on Windows 7 because that's just an awful experience all around and I can name dozens of other unpleasant things I'd rather be doing.
